server/weaverbird/steps/filter.py

```diff
@@ -11,17 +12,64 @@
 DataValue = Any  # FIXME use Any to prevent pydantic cast
 
 
-class SimpleCondition(BaseModel):
+class BaseSimpleCondition(BaseModel, ABC):
     column: ColumnName
-    # TODO support all operators
-    operator: Literal['eq']
+
+    @abstractmethod
+    def filter(self, df: DataFrame) -> Series:
+        ...
+
+
+class ComparisonCondition(BaseSimpleCondition):
+    operator: Literal['eq', 'ne', 'lt', 'le', 'gt', 'ge']
     value: DataValue
 
+    def filter(self, df: DataFrame) -> Series:
+        return getattr(df[self.column], self.operator)(self.value)
+
+
+class InclusionCondition(BaseSimpleCondition):
+    operator: Literal['in', 'nin']
+    value: List[DataValue]
+
+    def filter(self, df: DataFrame) -> Series:
+        f = df[self.column].isin(self.value)
+        if self.operator.startswith('n'):
+            return ~f
+        else:
+            return f
+
+
+class NullCondition(BaseSimpleCondition):
+    operator: Literal['null', 'notnull']
+
+    def filter(self, df: DataFrame) -> Series:
+        f = df[self.column].isnull()
+        if self.operator.startswith('not'):
+            return ~f
+        else:
+            return f
+
+
+class MatchCondition(BaseSimpleCondition):
+    operator: Literal['matches', 'notmatches']
+    value: str
+
+    def filter(self, df: DataFrame) -> Series:
+        f = df[self.column].str.match(self.value)
```

Comment:
`.str.match` regarde juste le début de la chaîne et pas toute la chaîne, est-ce que c'est ça qu'on veut ? Dans l'exemple des tests, `tat*` matche aussi bien `tata` que `taboulé`

Reply:
You're right, I switched to `str.contains` to match anywhere in the string.